### ISSUE TITLE
Very infrequent Error when Session Starts on IIS after Install

### ISSUE THREAD

**Issue by eSilverStrike** [Geeklog-Core/geeklog#1004]
Just wanted to record this error in case it becomes a bigger problem in the future. The odd time after an install I will get this E_NOTICE error on the success page about php sessions. It only happens once (ie refreshing the page the error doesn't happen again). This is using Geeklog v2.2.1 Beta 3 with the new session handling system (now using PHP built in session management)

It only happens when I am using IIS. The current PHP version installed on it is v7.0

Searching on the internet seems to indicate this is a IIS only issue.

```
A PHP error has occurred:
This is being displayed as "Root Debugging" is enabled in your Geeklog configuration.

If this is a production website you must disable this option once you have resolved any issues you are investigating.

E_NOTICE(8) - session_start(): ps_files_cleanup_dir: opendir(C:\Windows\temp) failed: No error (0) @ C:\inetpub\GitHub\geeklog\system\classes\Session.php line 282
Call Stack
# 	Function 	File 	Line
1 	session_start 	C:\inetpub\GitHub\geeklog\system\classes\Session.php 	282
2 	start 	C:\inetpub\GitHub\geeklog\system\classes\Session.php 	124
3 	init 	C:\inetpub\GitHub\geeklog\system\lib-sessions.php 	100
4 	SESS_sessionCheck 	C:\inetpub\GitHub\geeklog\public_html\lib-common.php 	272
5 	require_once 	C:\inetpub\GitHub\geeklog\public_html\admin\install\success.php 	42

array(0) {
}

```

**Comment by hostellerie** [comment]
I reviewed this issue again, and I am not sure this should still be considered a Geeklog Core bug.

The reported error comes from PHP's file-based session handler during `session_start()`:

```text
session_start(): ps_files_cleanup_dir:
opendir(C:\Windows\temp) failed
```

The report was specific to IIS with PHP 7.0, happened only very occasionally after installation, and disappeared after refreshing the page.

In the current Geeklog code, `Session::start()` still correctly calls `session_start()` and reacts if starting the session actually fails. I would not recommend suppressing the warning with something like:

```php
@session_start();
```

because that could also hide legitimate configuration problems such as an invalid `session.save_path`, missing permissions, or an unwritable session directory.

This looks more like a PHP/IIS session storage or garbage-collection issue than something Geeklog should work around in `Session.php`.

A possible improvement would instead be to add a diagnostic check to the installer. For example, Geeklog could inspect the configured PHP session storage and warn the administrator if the session directory is obviously missing, unreadable, or unwritable.

The installer could then provide a message explaining that `session.save_path` and the IIS/PHP permissions should be checked.

Considering that this issue was reported in 2019 with PHP 7.0, has no additional reproductions, and appears environment-specific, I think there are two reasonable options:

1. Close this issue as an IIS/PHP environment issue.
2. If we want to improve the installation experience, open a separate low-priority enhancement for checking PHP session storage configuration during installation.

I would avoid adding IIS-specific handling or suppressing session errors in the Geeklog session class itself.

**Comment by eSilverStrike** [comment]
I tend to agree. I'll close this issue now. I also have no real way to test this anything related to this since I have moved from IIS a while ago.
